### PR TITLE
24w34a snapshot support

### PR DESCRIPTION
**Pull request by Outfluencer**

### CODE REVIEW

File: protocol/src/main/java/net/md_5/bungee/protocol/packet/ClientSettings.java [hunk 1]
```diff
@@ -46,6 +47,10 @@ public void read(ByteBuf buf, ProtocolConstants.Direction direction, int protoco
         {
             allowServerListing = buf.readBoolean();
         }
+        if ( protocolVersion >= ProtocolConstants.MINECRAFT_1_21_2 )
+        {
+            particleStatus = ParticleStatus.values()[buf.readByte()];
```
Comment (FlorianMichael) [inline protocol/src/main/java/net/md_5/bungee/protocol/packet/ClientSettings.java:52]:
```suggestion
            particleStatus = ParticleStatus.values()[DefinedPacket.readVarInt( buf )];
```

File: protocol/src/main/java/net/md_5/bungee/protocol/packet/ClientSettings.java [hunk 2]
```diff
@@ -74,11 +79,22 @@ public void write(ByteBuf buf, ProtocolConstants.Direction direction, int protoc
         {
             buf.writeBoolean( allowServerListing );
         }
+        if ( protocolVersion >= ProtocolConstants.MINECRAFT_1_21_2 )
+        {
+            buf.writeByte( particleStatus.ordinal() );
```
Comment (FlorianMichael) [inline protocol/src/main/java/net/md_5/bungee/protocol/packet/ClientSettings.java:84]:
```suggestion
            DefinedPacket.writeVarInt( particleStatus.ordinal(), buf );
```

**Comment by FlorianMichael** [comment]
There are also missing packet id changes like (e.g serverbound chat packet), I would generally recommend to auto generate the Protocol enum rather than manual updating.

**Comment by Outfluencer** [comment]
> There are also missing packet id changes like (e.g serverbound chat packet), I would generally recommend to auto generate the Protocol enum rather than manual updating.

[24w33a_Clientbound_game.txt](https://github.com/user-attachments/files/16710418/24w33a_Clientbound_game.txt)
[24w33a_Serverbound_game.txt](https://github.com/user-attachments/files/16710419/24w33a_Serverbound_game.txt)

i generated this for the mappings now thank you

**Comment by Outfluencer** [comment]
testserver for the latest snapshot: snapshot.outfluencer.dev
If someone is interested

File: protocol/src/main/java/net/md_5/bungee/protocol/Protocol.java [hunk 3]
```diff
@@ -321,7 +327,7 @@ public enum Protocol
                     map( ProtocolConstants.MINECRAFT_1_19_4, 0x5F ),
                     map( ProtocolConstants.MINECRAFT_1_20_2, 0x61 ),
                     map( ProtocolConstants.MINECRAFT_1_20_3, 0x63 ),
-                    map( ProtocolConstants.MINECRAFT_1_20_5, 0x65 )
```
Comment (NEZNAMY) [inline protocol/src/main/java/net/md_5/bungee/protocol/Protocol.java:324]: Is this intended?

Comment (Outfluencer) [reply protocol/src/main/java/net/md_5/bungee/protocol/Protocol.java:324]: no i think its not thank you i will fix it

**Comment by md-5** [comment]
Thanks

**Pull request closed without merging**